tensorflow/lite/micro/kernels/cmsis-nn/mul.cc

```diff
@@ -62,8 +62,8 @@ TfLiteStatus CalculateOpData(TfLiteContext* context, TfLiteNode* node,
         context, params->activation, output, &data->output_activation_min,
         &data->output_activation_max));
 
-    double real_multiplier =
-        input1->params.scale * input2->params.scale / output->params.scale;
+    double real_multiplier = static_cast<double>(
+        input1->params.scale * input2->params.scale / output->params.scale);
```

Comment:
Want to match the micro reference kernel and do the computation in double precision?

https://github.com/tensorflow/tensorflow/blob/0b175ea29563da11dcfed3c70fbb93beff36f118/tensorflow/lite/micro/kernels/mul.cc#L67-L69

Lite is different:
https://github.com/tensorflow/tensorflow/blob/0b175ea29563da11dcfed3c70fbb93beff36f118/tensorflow/lite/kernels/mul.cc#L99-L100

I'll let you decide since the inconsistency in on our side, whatever you prefer to avoid issues like this one:
https://github.com/tensorflow/tensorflow/issues/42648

Reply:
Good catch! I think we should match the micro reference kernel.